tests/helper/tests/key_val_in_file.py

```diff
@@ -15,7 +22,10 @@ def _get_content_remote_file(client, fname):
 
 def _get_key_val_from_content(content, dict, fname):
     """ Validates if key/val are present in content """
+    present = False
     for k in dict:
         for line in content.split('\n'):
             if line.startswith(k):
-                assert dict[k] in line, f"Could not find {dict} in {fname}."
+                if dict[k] in line:
```

Comment:
In my opinion, this if condition is quiet dangerous. It either checks an exact match nor does it ignore the key of a specifc line.

Reply:
That's what this function should do; only validate if the value is somewhere present. While the exact match for the keys makes sense, only a general presence of the value somewhere by the given line should be done here.